chore(ui): Deprecate `output_text_verbatim()` with warning

karangattu opened the pull request:
Added a deprecation warning to the `output_text_verbatim()`

- [x] Merge PR: https://github.com/posit-dev/py-shiny-templates/pull/47
- [x] Merge PR: https://github.com/posit-dev/py-shiny-templates/pull/48

karangattu committed:
Deprecate output_text_verbatim with warning

Added a deprecation warning to the output_text_verbatim

karangattu committed:
Document deprecation of ui.output_text_verbatim()

`ui.output_text_verbatim()` is now deprecated in favor of `ui.output_text()` for text output and `ui.output_code()` for code output. This update adds a deprecation notice to the changelog.

karangattu committed:
Replace output_text_verbatim with output_code in UI

karangattu committed:
Replace output_text_verbatim with output_code in renderers

schloerke reviewed (changes requested):
Also adjust `@render.text` to `@render.code`

schloerke commented on shiny/ui/_output.py:
```diff
@@ -10,10 +10,12 @@
 
 from htmltools import Tag, TagAttrValue, TagFunction, css, div, tags
 
+from .._deprecated import ShinyDeprecationWarning
```
```suggestion
from .._deprecated import warn_deprecated
```

schloerke commented on shiny/ui/_output.py:
```diff
@@ -346,6 +348,14 @@ def output_text_verbatim(id: str, placeholder: bool = False) -> Tag:
+        "or `ui.output_code()` if you want to create an output container for code (monospaced text).",
+        category=ShinyDeprecationWarning,
+        stacklevel=2,
+    )
```
```suggestion
    warn_deprecated(
        "`ui.output_text_verbatim()` was deprecated in v1.6.0."
        "Please use `ui.output_text()` / `@render.text` to create an output container for plain text "
        "or `ui.output_code()` / `@render.code` to create an output container for monospaced text."
    )
```

schloerke reviewed (approved):
LGTM given comments / ✅ s

schloerke commented on shiny/ui/_output.py:
```suggestion
    Deprecated. Create a output container for some text.
```

schloerke commented on shiny/ui/_output.py:
Please also adjust any comment to point to `shiny.ui.output_code`

schloerke commented on shiny/ui/_output.py:
If appropriate, add `shiny.render.code`

schloerke reviewed (changes requested):
(See comments)

karangattu committed:
Switch output from render.text to render.code

Replaces @render.text with @render.code in multiple example, test, and template apps to standardize output for monospaced/code-style text. Updates documentation and deprecation warnings to reflect the change, and removes related test error allowances for deprecated output_text_verbatim usage.

karangattu committed:
Merge branch 'main' into add-deprecation-warnings-for-output_txt_verbatim

karangattu committed:
Update output code label and test expectation

Changed the output label from 'Output text verbatim content' to 'Output Code content' in the app and updated the corresponding test to match the new label and output type.

karangattu committed:
Merge branch 'add-deprecation-warnings-for-output_txt_verbatim' of https://github.com/posit-dev/py-shiny into add-deprecation-warnings-for-output_txt_verbatim

schloerke committed:
Merge branch 'main' into add-deprecation-warnings-for-output_txt_verbatim

schloerke committed:
Merge branch 'main' into add-deprecation-warnings-for-output_txt_verbatim

schloerke committed:
chore(ui): Fix the `output_text_verbatim()` deprecation message and docs

The deprecation notice named v1.6.0, but 1.7.0 has already shipped, so the
next release is v1.8.0. Also adds the missing space between the two sentences
of the message, drops the docstring cross-references to the deprecated
function from `output_text()` and `output_code()` (`output_code()`'s See Also
had come to reference itself), and covers the warning with a unit test.

schloerke committed:
chore: Migrate the remaining `output_text_verbatim()` call sites to `output_code()`

These examples, test apps, and the bundled testing skill were added to main
after this branch opened, so they would all emit the new deprecation warning.
Paired `@render.text` decorators become `@render.code`, and test controllers
become `OutputCode` (identical locator and API to `OutputTextVerbatim`, which
no longer has any app in the repo to target).

Drops tests/playwright/shiny/outputs/test_output_text_verbatim.py: its app
fixture is the `input_text` example, which now uses `output_code`, leaving the
test a duplicate of `test_input_text_app` with placeholder coverage already in
test_output_code.py.

schloerke committed:
chore(playwright): Deprecate the `OutputTextVerbatim` controller

`OutputTextVerbatim` only makes sense against `ui.output_text_verbatim()`, so it
is deprecated on the same schedule; constructing one now warns and points at
`OutputCode`, which has an identical locator and API.

Restores an end-to-end test of `ui.output_text_verbatim()` under
tests/playwright/shiny/deprecated/, with its own app so the deprecated output
container keeps its coverage until it is removed.

schloerke committed:
docs: Move the deprecated `output_text_verbatim` entries to Deprecated sections

`ui.output_text_verbatim` joins the existing Deprecated section in the core
reference, and the testing reference gains one at the bottom for
`playwright.controller.OutputTextVerbatim`, so neither sits alongside the
replacement it should not be confused with.

schloerke committed:
docs: Move the deprecated `render.download` to the Deprecated section

It is deprecated in favor of `render.download_button()` / `render.download_link()`
but was still listed beside them under "Rendering outputs".

schloerke committed:
docs: Note the undocumented, deprecated `ui.update_navs()` in the Deprecated section

It still exists but is intentionally left out of the reference, so record it as a
commented entry rather than leaving its absence look like an oversight.

schloerke committed:
docs(ui): Mark `output_text_verbatim()` as `@no_example()`

Its example was borrowed from `input_text`, which now demonstrates
`output_code()`, so the deprecated function's page was showing an example that
never called it. Matches `ui.panel_well()`, the other deprecated UI function.

schloerke committed:
docs(ui): Restore the `render.code` cross-reference on `output_text_verbatim()`

Requested in review; it was dropped when the See Also section was retargeted at
the replacement UI functions.

schloerke committed:
chore: Match the house style for the `output_text_verbatim` deprecation messages

Of the 19 deprecation messages in the package, only one states a version, and it
is an argument consolidation rather than a symbol. Follow the dominant
```X` is deprecated. Please use `Y` instead.`` shape instead, which also removes
a version string that would need bumping if the release slips.
